### PR TITLE
SRVKP-3342: Add pipelines cluster pool

### PR DESCRIPTION
**Pull request by divyansh42**
This PR adds the cluster pool and its required configurations. A follow-up PR will be done to modify the cluster claim test configurations of the different components.

### CODE REVIEW

File: clusters/hive/pools/pipelines/OWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+approvers:
+- divyansh42
```
Comment (khrm) [inline clusters/hive/pools/pipelines/OWNERS:2]: Add @jkandasa @savitaashture and me also.

Comment (divyansh42) [reply clusters/hive/pools/pipelines/OWNERS:2]: Done. 
I have added myself to the reviewers list.

**Comment by divyansh42** [comment]
@savitaashture @jkandasa @khrm  Can I please get review on this PR. Thanks.

**Review by khrm — COMMENTED** [review]
/lgtm

File: clusters/hive/pools/pipelines/pipelines-install-config.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pipelines-install-config
+  namespace: pipelines-cluster-pool
+type: Opaque
+stringData:
+  install-config.yaml: |-
+    apiVersion: v1
+    baseDomain: psi.ospqa.com
```
Comment (savitaashture) [inline clusters/hive/pools/pipelines/pipelines-install-config.yaml:10]: is this the same PSI cluster which pipelines team using today ?

File: clusters/hive/pools/pipelines/pipelines-install-config.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    compute:
+    - name: worker
+      platform:
+        openstack:
+          type: ci.m1.xlarge
+      replicas: 2
```
Comment (savitaashture) [inline clusters/hive/pools/pipelines/pipelines-install-config.yaml:16]: is this the minimum resource requirement  for master and worker

**Comment by savitaashture** [comment]
* Can we add one more cluster pool yaml for 4.16 as well because Pipeline supported on 4.16 also

File: clusters/hive/pools/pipelines/pipelines-install-config.yaml (hunk 2 above)
Comment (divyansh42) [reply clusters/hive/pools/pipelines/pipelines-install-config.yaml:10]: Yes

File: clusters/hive/pools/pipelines/pipelines-install-config.yaml (hunk 3 above)
Comment (divyansh42) [reply clusters/hive/pools/pipelines/pipelines-install-config.yaml:16]: @savitaashture I have a very limited idea on the resource requirements. Therefore, I haven't changed anything related to the resources. I have copied the configuration from the existing configuration used by the QE folks.
I am planning to change this based on the requirements. Please guide me if you think this needs to be changed.

**Comment by divyansh42** [comment]
> * Can we add one more cluster pool yaml for 4.16 as well because Pipeline supported on 4.16 also

Sure, will add for 4.16 as well.

File: clusters/hive/pools/pipelines/pipelines-install-config.yaml (hunk 3 above)
Comment (khrm) [reply clusters/hive/pools/pipelines/pipelines-install-config.yaml:16]: We will need more for the Performance team.

Comment (divyansh42) [reply clusters/hive/pools/pipelines/pipelines-install-config.yaml:16]: @khrm do you know how much that should be?

File: clusters/hive/pools/pipelines/OWNERS [hunk 4]
```diff
@@ -0,0 +1,10 @@
+approvers:
+- jkandasa
+- khrm
+- savitaashture
```
Comment (savitaashture) [inline clusters/hive/pools/pipelines/OWNERS:4]: Lets add @vdemeester as well under approvers

Comment (divyansh42) [reply clusters/hive/pools/pipelines/OWNERS:4]: Done

**Comment by divyansh42** [comment]
> 1.15 OSP supported from 4.14 onwards as per the component matrix http://dashboard.apps.cicd.ospqa.com/product/componentmatrix/ should we add clusterpool for only those versions ?? wdyt @khrm @jkandasa @vdemeester ?

I think that probably makes sense. 
When I was adding cluster pools for all the versions, I was under the impression that we would have test jobs for all other versions as well. But while modifying cluster pool in those test jobs, I only see 4.14 related tests.

**Comment by divyansh42** [comment]
@savitaashture Removed 4.12 and 4.13 cluster pool. PTAL.

**Review by vdemeester — APPROVED** [review]

**Comment by savitaashture** [comment]
Thank you @divyansh42 :tada:

**Comment by divyansh42** [comment]
As per the comment from the CI bot I am tagging @droslean for the approval. @droslean could you please approve the PR? Thank you.
/assign @droslean

File: clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-14-0-amd64-openstack-clusterpool.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  labels:
+    tp.openshift.io/owner: pipelines
+  platform:
+    openstack:
+      cloud: psi-pipelines
+      credentialsSecretRef:
+        name: pipelines-openstack-credentials
+  size: 0
+  skipMachinePools: true
+status:
```
Comment (deepsm007) [inline clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-14-0-amd64-openstack-clusterpool.yaml:33]: you can remove status

File: clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-15-0-amd64-openstack-clusterpool.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  labels:
+    tp.openshift.io/owner: pipelines
+  platform:
+    openstack:
+      cloud: psi-pipelines
+      credentialsSecretRef:
+        name: pipelines-openstack-credentials
+  size: 0
+  skipMachinePools: true
+status:
```
Comment (deepsm007) [inline clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-15-0-amd64-openstack-clusterpool.yaml:33]: same as above

File: clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-16-0-amd64-openstack-clusterpool.yaml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  labels:
+    tp.openshift.io/owner: pipelines
+  platform:
+    openstack:
+      cloud: psi-pipelines
+      credentialsSecretRef:
+        name: pipelines-openstack-credentials
+  size: 0
+  skipMachinePools: true
+status:
```
Comment (deepsm007) [inline clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-16-0-amd64-openstack-clusterpool.yaml:33]: same as above

File: clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-15-0-amd64-openstack-clusterpool.yaml (hunk 6 above)
Comment (divyansh42) [reply clusters/hosted-mgmt/hive/pools/pipelines/pipelines-ocp-4-15-0-amd64-openstack-clusterpool.yaml]: Done, thanks for the review @deepsm007 🙂

**Pull request merged**